cluster/juju/layers/kubernetes-master/actions/get-kubeconfig

```diff
@@ -0,0 +1,35 @@
+#!/usr/local/sbin/charm-env python3
+import os
+from yaml import safe_load as load
+from charmhelpers.core.hookenv import (
+    action_get,
+    action_set,
+    action_fail,
+    action_name
+)
+from charmhelpers.core.templating import render
+from subprocess import check_output
+
+
+def _kubectl(args):
+    """
+    Executes kubectl with args as arguments
+    """
+    snap_bin = os.path.join(os.sep, "snap", "bin")
+    env = os.environ.copy()
+    env["PATH"] = os.pathsep.join([snap_bin, env["PATH"]])
+    cmd = ["kubectl", "--kubeconfig=/home/ubuntu/config"]
+    cmd.extend(args)
+    return check_output(cmd, env=env)
+
+
+def get_kubeconfig():
+    """
+    Read the kubeconfig on the master and return it as JSON
+    """
+    kubeconfig = load(_kubectl(["config", "view", "-o", "yaml", "--raw"]))
+    action_set({"credential": kubeconfig})
```

Comment:
Passing the loaded data directly to `action_set` results in messy action output:

```
$ juju run-action kubernetes-master/0 get-kubeconfig --wait
unit-kubernetes-master-0:
  id: 9846184a-8766-495a-87f2-29ecd7ba18ad
  results:
    credential: '{''apiVersion'': ''v1'', ''clusters'': [{''cluster'': {''certificate-authority-data'':
      ''REDACTED'',
      ''server'': ''https://REDACTED:6443''}, ''name'': ''juju-cluster''}], ''contexts'':
      [{''context'': {''cluster'': ''juju-cluster'', ''user'': ''admin''}, ''name'':
      ''juju-context''}], ''current-context'': ''juju-context'', ''kind'': ''Config'',
      ''preferences'': {}, ''users'': [{''name'': ''admin'', ''user'': {''password'':
      ''REDACTED'', ''username'': ''admin''}}]}'
  status: completed
  timing:
    completed: 2019-03-19 21:56:08 +0000 UTC
    enqueued: 2019-03-19 21:56:07 +0000 UTC
    started: 2019-03-19 21:56:08 +0000 UTC
  unit: kubernetes-master/0
```

You can see the action output presenting the kubeconfig here, but it's a string that containers neither YAML nor JSON. It's a Python formatted string, which you can see the `action_set` function doing [here](https://github.com/juju/charm-helpers/blob/master/charmhelpers/core/hookenv.py#L964).

I recommend skipping the `yaml.load` and instead passing the raw kubeconfig string to `action_set`.

If it really needs to be JSON, then keep the `yaml.load`, but use `json.dumps` before you pass it to `action_set`.

Reply:
Thanks for the extensive test and great recommendations! Will rewrite this action and submit for review again.